Tests/Common/LoRaDeviceTwin.cs

```diff
@@ -0,0 +1,172 @@
+                SetDesiredPropertyIfExists(twin, TwinProperty.FCntDownStart, someDesiredProperties.FCntDownStart?.ToString(CultureInfo.InvariantCulture));
+                SetDesiredPropertyIfExists(twin, TwinProperty.FCntResetCounter, someDesiredProperties.FCntResetCounter?.ToString(CultureInfo.InvariantCulture));
+                SetDesiredPropertyIfExists(twin, TwinProperty.RX1DROffset, someDesiredProperties.Rx1DROffset?.ToString(CultureInfo.InvariantCulture));
+                SetDesiredPropertyIfExists(twin, TwinProperty.RX2DataRate, someDesiredProperties.Rx2DataRate?.ToString());
+                var receiveWindow = someDesiredProperties.PreferredWindow switch
+                {
+                    ReceiveWindowNumber.ReceiveWindow1 => 1,
+                    ReceiveWindowNumber.ReceiveWindow2 => 2,
+                    _ => (int?)null
+                };
+                SetDesiredPropertyIfExists(twin, TwinProperty.PreferredWindow, receiveWindow?.ToString(CultureInfo.InvariantCulture));
+                SetDesiredPropertyIfExists(twin, TwinProperty.RXDelay, someDesiredProperties.RxDelay?.ToString(CultureInfo.InvariantCulture));
+                SetDesiredPropertyIfExists(twin, TwinProperty.ClassType, someDesiredProperties.ClassType?.ToString());
+                SetDesiredPropertyIfExists(twin, TwinProperty.KeepAliveTimeout, someDesiredProperties.KeepAliveTimeout?.TotalSeconds.ToString(CultureInfo.InvariantCulture));
+            }
+
+            if (reportedProperties is { } someReportedProperties)
+            {
+                SetReportedPropertyIfExists(twin, TwinProperty.FCntUp, someReportedProperties.FCntUp?.ToString(CultureInfo.InvariantCulture));
+                SetReportedPropertyIfExists(twin, TwinProperty.FCntUpStart, someReportedProperties.FCntUpStart?.ToString(CultureInfo.InvariantCulture));
+                SetReportedPropertyIfExists(twin, TwinProperty.FCntDown, someReportedProperties.FCntDown?.ToString(CultureInfo.InvariantCulture));
+                SetReportedPropertyIfExists(twin, TwinProperty.FCntDownStart, someReportedProperties.FCntDownStart?.ToString(CultureInfo.InvariantCulture));
+                SetReportedPropertyIfExists(twin, TwinProperty.AppSKey, someReportedProperties.AppSessionKey?.ToString());
+                SetReportedPropertyIfExists(twin, TwinProperty.NwkSKey, someReportedProperties.NetworkSessionKey?.ToString());
+                SetReportedPropertyIfExists(twin, TwinProperty.DevAddr, someReportedProperties.DevAddr?.ToString());
+                SetReportedPropertyIfExists(twin, TwinProperty.DevNonce, someReportedProperties.DevNonce?.ToString());
+                SetReportedPropertyIfExists(twin, TwinProperty.FCntResetCounter, someReportedProperties.FCntResetCounter?.ToString(CultureInfo.InvariantCulture));
+                SetReportedPropertyIfExists(twin, TwinProperty.PreferredGatewayID, someReportedProperties.PreferredGatewayId?.ToString());
```

Comment:
This is already a string so conversion is redundant:

```suggestion
                SetReportedPropertyIfExists(twin, TwinProperty.PreferredGatewayID, someReportedProperties.PreferredGatewayId);
```